pkg/hubble/parser/seven/http_test.go

```diff
@@ -140,6 +141,126 @@ func TestDecodeL7HTTPRequest(t *testing.T) {
 	assert.Equal(t, "4bf92f3577b34da6a3ce929d0e0e4736", f.GetTraceContext().GetParent().GetTraceId())
 }
 
+func TestDecodeL7HTTPRequestRemoveUrlQuery(t *testing.T) {
+	requestPath, err := url.Parse("http://myhost/some/path#v2?foo=bar")
+	require.NoError(t, err)
+	lr := &accesslog.LogRecord{
+		Type:                accesslog.TypeRequest,
+		Timestamp:           fakeTimestamp,
+		NodeAddressInfo:     fakeNodeInfo,
+		ObservationPoint:    accesslog.Ingress,
+		SourceEndpoint:      fakeSourceEndpoint,
+		DestinationEndpoint: fakeDestinationEndpoint,
+		IPVersion:           accesslog.VersionIPv4,
+		Verdict:             accesslog.VerdictForwarded,
+		TransportProtocol:   accesslog.TransportProtocol(u8proto.TCP),
+		ServiceInfo:         nil,
+		DropReason:          nil,
+		HTTP: &accesslog.LogRecordHTTP{
+			Code:     0,
+			Method:   "POST",
+			URL:      requestPath,
+			Protocol: "HTTP/1.1",
+			Headers: http.Header{
+				"Host":        {"myhost"},
+				"Traceparent": {"00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01"},
+			},
+		},
+	}
+	lr.SourceEndpoint.Port = 56789
+	lr.DestinationEndpoint.Port = 80
+
+	dnsGetter := &testutils.FakeFQDNCache{
+		OnGetNamesOf: func(epID uint32, ip netip.Addr) (names []string) {
+			ipStr := ip.String()
+			switch {
+			case epID == uint32(fakeSourceEndpoint.ID) && ipStr == fakeDestinationEndpoint.IPv4:
+				return []string{"endpoint-1234"}
+			case epID == uint32(fakeDestinationEndpoint.ID) && ipStr == fakeSourceEndpoint.IPv4:
+				return []string{"endpoint-4321"}
+			}
+			return nil
+		},
+	}
+	IPGetter := &testutils.FakeIPGetter{
+		OnGetK8sMetadata: func(ip netip.Addr) *ipcache.K8sMetadata {
+			if ip == netip.MustParseAddr(fakeDestinationEndpoint.IPv4) {
+				return &ipcache.K8sMetadata{
+					Namespace: "default",
+					PodName:   "pod-1234",
+				}
+			}
+			return nil
+		},
+	}
+	serviceGetter := &testutils.FakeServiceGetter{
+		OnGetServiceByAddr: func(ip netip.Addr, port uint16) *flowpb.Service {
+			if ip == netip.MustParseAddr(fakeDestinationEndpoint.IPv4) && (port == fakeDestinationEndpoint.Port) {
+				return &flowpb.Service{
+					Name:      "service-1234",
+					Namespace: "default",
+				}
+			}
+			return nil
+		},
+	}
+	endpointGetter := &testutils.FakeEndpointGetter{
+		OnGetEndpointInfo: func(ip netip.Addr) (endpoint v1.EndpointInfo, ok bool) {
+			switch {
+			case ip == netip.MustParseAddr(fakeSourceEndpoint.IPv4):
+				return &testutils.FakeEndpointInfo{
+					ID: fakeSourceEndpoint.ID,
+				}, true
+			case ip == netip.MustParseAddr(fakeDestinationEndpoint.IPv4):
+				return &testutils.FakeEndpointInfo{
+					ID: fakeDestinationEndpoint.ID,
+				}, true
+			}
+			return nil, false
+		},
+	}
+
+	opts := []options.Option{options.Redact(nil, []string{"http-url-query"})}
+	parser, err := New(log, dnsGetter, IPGetter, serviceGetter, endpointGetter, opts...)
```

Comment:
It doesn't feel right to duplicate all the unrelated checks here. I think we could pass `nil` for `dnsGetter`, `IPGetter`, `serviceGetter`, and `endpointGetter` and check only the HTTP related fields in the decoded flow.